manager/state/store/by.go

```diff
@@ -143,3 +143,51 @@ func (b byReferencedSecretID) isBy() {
 func ByReferencedSecretID(secretID string) By {
 	return byReferencedSecretID(secretID)
 }
+
+type byKind string
+
+func (b byKind) isBy() {
+}
+
+// ByKind creates an object to pass to Find to search for a Resource of a
+// particular kind.
+func ByKind(kind string) By {
+	return byKind(kind)
+}
+
+type byCustom struct {
+	objType string
+	index   string
+	value   string
+}
+
+func (b byCustom) isBy() {
+}
+
+// ByCustom creates an object to pass to Find to search a custom index.
+func ByCustom(objType, index, value string) By {
+	return byCustom{
+		objType: objType,
+		index:   index,
+		value:   value,
+	}
+}
+
+type byCustomPrefix struct {
```

Comment:
It has the same values but it's a different type. We type switch on these to figure out what action to perform. In this case, it will do a prefix match instead of an exact match.